### PR TITLE
Refactor ItemService unit tests #346

### LINKED ISSUE #346: Refactor items service unit tests
As part of #90 following on from #345.

### PR DESCRIPTION
**Pull request by joelvdavies**
## Description
See #346.

### Notes
- Removed `test_create_with_invalid_catalogue_category_id` and `test_update_with_invalid_usage_status` - we don't normally do invalid id checks in service tests.
- Removed `test_get_with_non_existent_id` - Already asserting return value, doesn't matter if its None or an item.
- Removed some update properties tests that were effectively testing `utils.process_properties` e.g. `test_update_change_property_value` - instead added test on the merging of properties instead and assert its called with the right parameters
- There are a few TODO's left that I have put where I pretty much copied things from catalogue items - we should be able to put them in a common place although its not clear to me how at the moment

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking
Closes #346

### CODE REVIEW

File: test/unit/services/test_item.py [hunk 1]
```diff
@@ -3,1304 +3,884 @@
...
-                **{
-                    **ITEM_INFO,
-                    "properties": add_ids_to_properties(None, ITEM_INFO["properties"]),
-                },
+        ServiceTestHelpers.mock_get(self.mock_catalogue_category_repository, self._catalogue_category_out)
+
+        # Catalogue item
+
+        # When properties are given need to add any property `id`s and ensure the expected data inserts them as well
+        catalogue_item_property_post_schemas = []
```
Comment (VKTB) [inline test/unit/services/test_item.py:154]:
```suggestion
```

File: test/unit/services/test_item.py [hunk 2]
```diff
@@ -3,1304 +3,884 @@
...
+    """Base class for `create` tests."""
+
+    # pylint:disable=too-many-instance-attributes
+    _catalogue_item_out: Optional[CatalogueItemOut]
+    _catalogue_category_out: Optional[CatalogueCategoryOut]
+    _usage_status_out: Optional[UsageStatusOut]
+    _item_post: ItemPostSchema
+    _expected_item_in: ItemIn
+    _expected_item_out: ItemOut
+    _created_item: CatalogueItemOut
```
Comment (VKTB) [inline test/unit/services/test_item.py:94]:
```suggestion
    _created_item: ItemOut
```

File: test/unit/services/test_item.py [hunk 3]
```diff
@@ -3,1304 +3,884 @@
...
-        ),
-    )
-
-    catalogue_item_id = str(ObjectId())
-    with pytest.raises(InvalidActionError) as exc:
-        item_service.update(
-            item.id,
-            ItemPatchSchema(catalogue_item_id=catalogue_item_id),
+class TestCreate(CreateDSL):
+    """Tests for creating a item."""
```
Comment (VKTB) [inline test/unit/services/test_item.py:280]:
```suggestion
    """Tests for creating an item."""
```

File: test/unit/services/test_item.py [hunk 4]
```diff
@@ -3,1304 +3,884 @@
...
+        :param message: Expected message of the raised exception.
+        """
+
+        self.mock_item_repository.update.assert_not_called()
+
+        assert str(self._update_exception.value) == message
+
+
+class TestUpdate(UpdateDSL):
+    """Tests for updating a catalogue item."""
```
Comment (VKTB) [inline test/unit/services/test_item.py:724]:
```suggestion
    """Tests for updating an item."""
```

**Review by VKTB — COMMENTED** [review]
Just some very minor things, it looks good otherwise.

File: test/unit/services/test_item.py (hunk 1 above)
Comment (joelvdavies) [reply test/unit/services/test_item.py:154]: Its used below.

Comment (VKTB) [reply test/unit/services/test_item.py:154]: Sorry I should have put a note. It is used but only inside that `IF` statement. What you currently have on line 157 should be just enough.

Comment (joelvdavies) [reply test/unit/services/test_item.py:154]: Oh right, yeah no sorry I didn't realise 😄

**Review by VKTB — APPROVED** [review]

**Pull request merged**